UnitTestProject1/UnitTestProject1/Task_20.cs

```diff
@@ -0,0 +1,173 @@
+        }
+
+        [TestMethod]
+        [TestCategory("Chrome")]
+        public void LogoutTest()
+        {
+            Login("EugenBorisik", "qwerty12345");
+            driver.FindElement(By.ClassName("sign-out")).Click();
+            string CurrentURL = driver.Url;
+            Assert.AreEqual(URL, CurrentURL);
+        }
+
+        [TestMethod]
+        [TestCategory("Chrome")]
+        public void LoginTestWithWaiter()
+        {
+            Login("EugenBorisik", "qwerty12345");
+            string CurrentURL = driver.Url;
+            Assert.AreEqual(ExpectedURL, CurrentURL);
+            driver.FindElement(By.Id("officeMenu")).Click();
+            string CurrentOfficeURL = driver.Url;
+            WebDriverWait wait = new WebDriverWait(driver, TimeSpan.FromSeconds(15));
+            wait.PollingInterval = TimeSpan.FromMilliseconds(2700);
+            wait.Until(ExpectedConditions.ElementIsVisible(By.Id("input-search")));
+            IWebElement Search = driver.FindElement(By.Id("input-search"));
+            Assert.AreEqual(OfficeTabURL, CurrentOfficeURL);
+            Assert.IsTrue(Search.Displayed, "Search input is not displayed!");
+         }
+
+        [TestMethod]
+        [DeploymentItem(@"SolutionItems\TestData.csv")]
+        [DataSource("Microsoft.VisualStudio.TestTools.DataSource.CSV", "TestData.csv", "TestData#csv", DataAccessMethod.Sequential)]
+        public void DDT_LoginTest()
+        {
+           string UName = TestContext.DataRow["UserName"].ToString();
+           string PWD = TestContext.DataRow["Password"].ToString();
+           Login(UName, PWD);
+           string CurrentURL = driver.Url;
+           Assert.AreEqual(ExpectedURL, CurrentURL);
+        }
+
+        [TestMethod]
+        [TestCategory("Chrome")]
+        public void FrameTest()
+        {
+           driver.Navigate().GoToUrl(FrameUrl);
+           IWebElement FrameLoc = driver.FindElement(By.Id("mce_0_ifr"));
+           driver.SwitchTo().Frame(FrameLoc);
+           driver.FindElement(By.Id("tinymce")).Clear();
+           driver.FindElement(By.Id("tinymce")).SendKeys("Hello ");
+           driver.FindElement(By.Id("tinymce")).SendKeys(KeyCode);
+           driver.FindElement(By.Id("tinymce")).SendKeys("World!");
+           driver.FindElement(By.Id("tinymce")).SendKeys(KeyCode);
+           string CurText = driver.FindElement(By.TagName("p")).Text;
+           Assert.IsTrue(CurText.Contains("World!") && CurText.Contains("Hello"));
+        }
+
+        [TestMethod]
+        [TestCategory("Chrome")]
+        public void JSAlert()
+        {
+            driver.Navigate().GoToUrl(AlertUrl);
+            driver.FindElement(By.XPath("//button[@onclick='jsAlert()']")).Click();
+            string AlertText = driver.SwitchTo().Alert().Text;
+            Assert.AreEqual(AlertText, AlertMSG);
+        }
+
+        [TestMethod]
+        [TestCategory("Chrome")]
+        public void JSConfirm()
```

Comment:
Split test into 2, because you verify different flows, and if first flow ('cancel' case) fails, the second one will be not performed at all